PDF manual: use microtype package, don't use ae package

xavierleroy opened the pull request:
The `microtype` package is used to turn ligatures off in typewriter font, fixing #10513.  It also promises  "subliminal refinements towards typographical perfection", which is too good to pass.

The `ae` package is shadowed by the `lmodern` package used later.

xavierleroy committed:
Disable ligatures in typewriter fonts

Because `--help` should be rendered with two minus signs, not one dash.

Use the "microtype" package for this purpose.

Fixes: #10513

wikku commented:
I can't see any adverse side effects except compilation time increasing by 15-20%. (I was worried about a subliminal sense of unease from `microtype` slightly shifting lines of monospace code, but according to the manual loading `listings` prevents it)  

There are positive effects of microtype: the margins look more even and there are fewer Underfull/Overfull warnings.

Octachron commented on manual/src/manual.tex:
```diff
@@ -20,13 +21,16 @@
 \input{macros.tex}
+
+% No ligatures in typewriter font
+\DisableLigatures{encoding = T1, family = tt* }
```
Hevea doesn't support `\DisableLigatures` (but renders correctly `--help`). It is probably better to define a noop `\DeclareLigatures` macro in `macros.hva` (around `\DeclareUnicodeCharacters`?).

xavierleroy replied:
Right.  Done.

xavierleroy commented:
>  compilation time increasing by 15-20%

Good point.  This is for building the PDF manual, I believe.  The total build time for the manual (incl. HTML and plain text versions) increases by 4-5%.  I think we can tolerate that.  (The manual is not built during OPAM switch installs, nor during most CI runs.)

xavierleroy committed:
Remove `\usepackage{ae}`, redundant with `\usepackage{lmodern}`

`ae` is a T1 encoding on top of the historical CM fonts.
`lmodern` is a more modern and more complete set of CM fonts,
  with T1 encoding from the beginning.

Using `ae` then `lmodern` is like using `lmodern`.
(The PDFs use exactly the same LM* fonts.)

Octachron commented:
I agree that a small increase of the manual build time is completely fine.

Octachron commented:
I am adding this PR to the 4.13.0 milestone in order to not forget it in my manual pass during the beta, but there is no urgency in cherry-picking it to 4.13 branch.
